security/integration/main.go

```diff
@@ -208,42 +317,59 @@ func deployIstioCA(clientset kubernetes.Interface) {
 		},
 	}
 	if _, err := clientset.RbacV1beta1().RoleBindings(opts.namespace).Create(&rolebinding); err != nil {
-		glog.Fatalf("failed to create rolebinding (error: %v)", err)
-	}
-	// Create Deployment
-	envVar := v1.EnvVar{
-		Name: "NAMESPACE",
-		ValueFrom: &v1.EnvVarSource{
-			FieldRef: &v1.ObjectFieldSelector{
-				APIVersion: "v1",
-				FieldPath:  "metadata.namespace",
-			},
-		},
+		return err
 	}
-	container := v1.Container{
-		Env:   []v1.EnvVar{envVar},
-		Name:  "istio-ca-container",
-		Image: fmt.Sprintf("%v/%v:%v", opts.containerHub, opts.containerImage, opts.containerTag),
+
+	return nil
+}
+
+func deployIstioCA(clientset kubernetes.Interface) {
+	// Create Role
+	err := createRole(clientset)
+	if err != nil {
+		glog.Fatal("failed to create a role (error: %v)", err)
 	}
-	spec := v1.PodSpec{
-		Containers: []v1.Container{container},
+
+	// Create RoleBinding
+	err = createRoleBinding(clientset)
+	if err != nil {
+		glog.Fatal("failed to create a rolebinding (error: %v)", err)
 	}
-	uuid := string(uuid.NewUUID())
-	objectMeta := metav1.ObjectMeta{
-		Labels: map[string]string{"uuid": uuid},
-		Name:   "istio-ca",
+
+	// Create Istio CA with self signed certificates
+	_, err = createPod(clientset,
+		fmt.Sprintf("%v/istio-ca:%v", opts.containerHub, opts.containerTag),
+		"istio-ca-self")
+	if err != nil {
+		glog.Fatalf("failed to deploy Istio CA (error: %v)", err)
 	}
-	pod := &v1.Pod{
-		ObjectMeta: objectMeta,
-		Spec:       spec,
+
+	// Create a Istio CA and Service with generated certificates
```

Comment:
Why do we test two CAs at the same time? Will they affect each other? I think we need to test the self-signed CA first, which is the old e2e test, and then clean up. After that, run the CA for node agent test case. What do you think?

Reply:
Separated the test into runSelfSignedCATests and runCAForNodeAgentTests